src/components/tags/TagList.js

```diff
@@ -4,39 +4,58 @@ import { TagContext } from "./TagProvider"
 import "./TagList.css"
 
 export const TagList = () => {
- const { tags, getTags } = useContext(TagContext)
- const loggedInUser = localStorage.getItem("lu_token")
+ const { tags, getTags, deleteTagById } = useContext(TagContext)
+ const loggedInUser = localStorage.getItem("rare_user_id")
 
  useEffect(() => {
   getTags()
  }, [])
 
+ const deleteHandler = ( indx ) => deleteTagById(indx)
+
  // So we wouldn't have to worry about missing ?'s in the return component
  // and avoid the "cannot find label of undefined" error.
  // Comment to commit and push for pull request.
- if(!tags) return (<div>Loading Tags</div>)
+ // Array.isArray(tags) - for some rason, component is keeping state after navigating to a different
+ // page and then navigating back. This can cause an error if Arra.isArray(tags) is replaced by just
+ // tags. To recreate the error, log out and clear cache/user token.
+ // Replace Array.isArray(tags) with tags.
+ // Click on the Tags Mangement link.
+ // Click on Login link and log in.
+ // Click on Tags Mangement link again.
+ if(!Array.isArray(tags)) return (<div>Loading Tags</div>)
 
  return (
-  tags
-  ?<>
-   <h2>Labels</h2>
-   <ul className="tags_list">
-     {
-      tags.map((tag, i) => <li key={i} className="tags_list--item">{ tag.label }</li>)
-     }
-   </ul>
+    loggedInUser && Array.isArray(tags)
+    ?
+      <>
+        <h2>Labels</h2>
+
+        <ul className="tags_list">
+          {
+            tags.map((tag) => {
 
-     {
-       loggedInUser
-       ?
+              return <li key={tag.id} className="tags_list--item">
+                <div className="tag_label">{ tag.label }</div>
+                { tag.isStaff && <div>
```

Comment:
When my client requests tags from the server, the objects don't have a property of `isStaff`. Is this an error for me only, or is something missing from the code?

Reply:
@Daniel-L-Ross  It was was the latest main branch. isStaff is set on Login.js